File: sdks/ts/src/sdk/models/shared/shieldinformationbarrier.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { UserBase } from "./userbase";
import { EnterpriseBase } from "./enterprisebase";


export enum ShieldInformationBarrierStatusEnum {
    Draft = "draft",
    Pending = "pending",
    Disabled = "disabled",
    Enabled = "enabled",
    Invalid = "invalid"
}

export enum ShieldInformationBarrierTypeEnum {
    ShieldInformationBarrier = "shield_information_barrier"
}


// ShieldInformationBarrier
/** 
 * A standard representation of a
 * shield information barrier object
**/
export class ShieldInformationBarrier extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=created_at" })
  createdAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=created_by" })
  createdBy?: UserBase;

  @SpeakeasyMetadata({ data: "json, name=enabled_at" })
  enabledAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=enabled_by" })
  enabledBy?: UserBase;

  @SpeakeasyMetadata({ data: "json, name=enterprise" })
  enterprise?: EnterpriseBase;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: ShieldInformationBarrierStatusEnum;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: ShieldInformationBarrierTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=updated_at" })
  updatedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=updated_by" })
  updatedBy?: UserBase;
}

File: python-sdk/src/boxsdk/models/shared/shieldinformationbarrier.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import user_base as shared_user_base
from ..shared import enterprise_base as shared_enterprise_base

class ShieldInformationBarrierStatusEnum(str, Enum):
    DRAFT = "draft"
    PENDING = "pending"
    DISABLED = "disabled"
    ENABLED = "enabled"
    INVALID = "invalid"

class ShieldInformationBarrierTypeEnum(str, Enum):
    SHIELD_INFORMATION_BARRIER = "shield_information_barrier"


@dataclass_json
@dataclasses.dataclass
class ShieldInformationBarrier:
    r"""ShieldInformationBarrier
    A standard representation of a
    shield information barrier object
    """
    
    created_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    created_by: Optional[shared_user_base.UserBase] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_by') }})
    enabled_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('enabled_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    enabled_by: Optional[shared_user_base.UserBase] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('enabled_by') }})
    enterprise: Optional[shared_enterprise_base.EnterpriseBase] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('enterprise') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    status: Optional[ShieldInformationBarrierStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    type: Optional[ShieldInformationBarrierTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    updated_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('updated_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    updated_by: Optional[shared_user_base.UserBase] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('updated_by') }})
    
